### PR TITLE
Example/dcip inversions

### PR DESCRIPTION
**Pull request by thast**
- 2D DC inversion: conductive and resistive cylinders
- 3D DC inversion: conductive and resistive spheres
- Update to DCUtils.gen_DCIPsurvey() : Survey Generator can be used now for 2.5D survey.

### CODE REVIEW

File: SimPEG/EM/Static/Utils/StaticUtils.py [hunk 1]
```diff
@@ -185,15 +185,11 @@ def gen_DCIPsurvey(endl, mesh, surveyType, a, b, n):
         :object mesh -> SimPEG mesh object
         :switch surveyType -> "dipole-dipole" (dipole-dipole) | "pole-dipole" (pole-dipole) | 'gradient'
         : param a, n -> pole seperation, number of rx dipoles per tx
+        :str d2flag: With a 2D mesh, choose between a full 2D problem ('2D') or a 2.5D problem ('2.5D')
```
Comment (lheagy) [inline SimPEG/EM/Static/Utils/StaticUtils.py]: I think this should be `:param str d2flag:` ?

File: SimPEG/EM/Static/Utils/StaticUtils.py [hunk 2]
```diff
@@ -261,19 +257,22 @@ def xy_2_r(x1, x2, y1, y2):
 
             # Create receiver poles
 
-            if mesh.dim==3:
+            if mesh.dim == 3:
```
Comment (lheagy) [inline SimPEG/EM/Static/Utils/StaticUtils.py]: thanks @thast!

File: examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    \\sigma = \\mathcal{M}(\\mathbf{m})
+
+Following example will show you how user can implement a 2D DC inversion.
+"""
+
+from SimPEG import (
+    Mesh,  Problem,  Survey,  Maps,  Utils,
+    EM,  DataMisfit,  Regularization,  Optimization,
+    InvProblem,  Directives,  Inversion
+    )
```
Comment (lheagy) [inline examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py]: would you mind closing this bracket at the beginning of the line? (eg. no indentation?)

File: examples/06-dc/TwoSpheres_DipoleDipole_3Dinversion.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    \\sigma = \\mathcal{M}(\\mathbf{m})
+
+Following example will show you how user can implement a 3D DC inversion.
+"""
+
+from SimPEG import (
+    Mesh,  Problem,  Survey,  Maps,  Utils,
+    EM,  DataMisfit,  Regularization,  Optimization,
+    InvProblem,  Directives,  Inversion
+    )
```
Comment (lheagy) [inline examples/06-dc/TwoSpheres_DipoleDipole_3Dinversion.py]: same comment re indentation

File: examples/06-dc/TwoSpheres_DipoleDipole_3Dinversion.py [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+from SimPEG.EM.Static import DC, Utils as DCUtils
+import numpy as np
+import matplotlib.pyplot as plt
+from pymatsolver import PardisoSolver
+
+np.random.seed(12345)
```
Comment (lheagy) [inline examples/06-dc/TwoSpheres_DipoleDipole_3Dinversion.py]: ha :)

File: examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
+"""
+2D DC inversion of Dipole Dipole array
```
Comment (lheagy) [inline examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py]: for the file name, would you mind using lowercase? and probably starting with `plot` (seems to be a bit of a convention with the examples cc @rowanc1). Maybe
`plot_dipoledipole_2Dinversion_twocylinders.py`

File: examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+from SimPEG import (
+    Mesh,  Problem,  Survey,  Maps,  Utils,
+    EM,  DataMisfit,  Regularization,  Optimization,
+    InvProblem,  Directives,  Inversion
+    )
+from SimPEG.EM.Static import DC, Utils as DCUtils
+import numpy as np
+import matplotlib.pyplot as plt
+from pymatsolver import PardisoSolver
```
Comment (lheagy) [inline examples/06-dc/TwoCylinders_DipoleDipole_2Dinversion.py]: We might want to put this under a `try`
```python
try:
    from pymatsolver import Pardiso as Solver
except ImportError:
    from SimPEG import SolverLU as Solver
```

**Comment by thast** [comment]
Thanks @lheagy !

File: SimPEG/EM/Static/Utils/StaticUtils.py [hunk 8]
```diff
@@ -184,16 +184,12 @@ def gen_DCIPsurvey(endl, mesh, surveyType, a, b, n):
         :param endl -> input endpoints [x1, y1, z1, x2, y2, z2]
         :object mesh -> SimPEG mesh object
         :switch surveyType -> "dipole-dipole" (dipole-dipole) | "pole-dipole" (pole-dipole) | 'gradient'
-        : param a, n -> pole seperation, number of rx dipoles per tx
+        :param a, n -> pole seperation, number of rx dipoles per tx
```
Comment (lheagy) [inline SimPEG/EM/Static/Utils/StaticUtils.py]: I realize that you didn't author this, but @thast, would you be willing to update this docstring so it follows standard sphinx? 
```
:param int a: pole seperation 
```

File: examples/06-dc/plot_dipoledipole_2Dinversion_twocylinders.py [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+problem = DC.Problem3D_CC(mesh,  sigmaMap=mapping)
+problem.pair(survey)
+problem.Solver = Solver
+
+survey.dpred(mtrue[actind])
+survey.makeSyntheticData(mtrue[actind], std=0.05, force=True)
+
+
+#####################
+# Tikhonov Inversion#
```
Comment (lheagy) [inline examples/06-dc/plot_dipoledipole_2Dinversion_twocylinders.py]: A bit picky, but for clarity, would you mind putting a space after "Inversion"?

**Comment by thast** [comment]
@lheagy : Done!

**Comment by lheagy** [comment]
Thanks @thast!

**Pull request merged**